vault_service/tests/test_bumblebee.py

```diff
@@ -9,59 +9,60 @@
 if project_home not in sys.path:
     sys.path.insert(0, project_home)
 
-from myads_service import app            
-from myads_service.models import db, Query, Institute, Library
+from vault_service import app
+from vault_service.models import Query, Institute, Library, Base
 
 class TestSite(TestCase):
     '''Tests that each route is an http response'''
-    
+
     def create_app(self):
         '''Start the wsgi application'''
         a = app.create_app(**{
-               'SQLALCHEMY_BINDS': {'myads': 'sqlite:///'},
+               'SQLALCHEMY_DATABASE_URI': 'sqlite:///',
                'SQLALCHEMY_ECHO': True,
                'TESTING': True,
                'PROPAGATE_EXCEPTIONS': True,
                'TRAP_BAD_REQUEST_ERRORS': True,
-               'MYADS_BUMBLEBEE_OPTIONS': {'foo': 'bar'}
+               'VAULT_BUMBLEBEE_OPTIONS': {'foo': 'bar'}
             })
-        db.create_all(app=a, bind=['myads'])
+        a.db.create_all()
         return a
 
 
     def test_store_data(self):
         '''Tests the ability to query site config'''
-        
+
         r = self.client.get(url_for('bumblebee.configuration'),
                 content_type='application/json')
         self.assertStatus(r, 200)
         self.assert_(r.json == {'foo': 'bar'}, 'missing json response')
-        
+
         r = self.client.get(url_for('bumblebee.configuration') + '/foo',
                 content_type='application/json')
         self.assertStatus(r, 200)
         self.assert_(r.json == 'bar', 'missing json response')
-        
+
         r = self.client.get(url_for('bumblebee.configuration') + '/foox',
                 content_type='application/json')
         self.assertStatus(r, 404)
 
 class TestOpenURL(TestCase):
     '''Tests that each route is an http response'''
-    
+
     def create_app(self):
         '''Start the wsgi application'''
         a = app.create_app(**{
-               'SQLALCHEMY_BINDS': {'myads': 'sqlite:///', 'institutes': 'sqlite:///'},
+               'SQLALCHEMY_DATABASE_URI': 'sqlite:///',
                'SQLALCHEMY_ECHO': True,
                'TESTING': True,
                'PROPAGATE_EXCEPTIONS': True,
                'TRAP_BAD_REQUEST_ERRORS': True,
-               'MYADS_BUMBLEBEE_OPTIONS': {'foo': 'bar'}
+               'VAULT_BUMBLEBEE_OPTIONS': {'foo': 'bar'}
             })
-        db.create_all(app=a, bind=['institutes'])
+        Base.query = a.db.session.query_property()
```

Comment:
why not `a.db.create_all()` as you did above?

Reply:
Actually I have reorganised this a bit (I had some duplications and I was not using test setUp/tearDown consistently). The `a.db.create_all()` is not creating the tables it should create while `Base.metadata.create_all(bind=self.app.db.engine)` does it. I don't fully understand why a.db does not know about all the defined models, but using Base.metadata we are sure it will create all the tables we need.